src/pytorch_lightning/callbacks/model_checkpoint.py

```diff
@@ -592,14 +591,10 @@ def __resolve_ckpt_dir(self, trainer: "pl.Trainer") -> None:
             # the user has changed weights_save_path
             ckpt_path = os.path.join(trainer._weights_save_path_internal, "checkpoints")
         elif trainer.loggers:
-            if len(trainer.loggers) == 1:
-                assert trainer.logger is not None
-                save_dir = trainer.logger.save_dir or trainer.default_root_dir
-            else:
-                save_dir = trainer.default_root_dir
-
-            name = _name(trainer.loggers)
-            version = _version(trainer.loggers)
+            assert trainer.loggers[0] is not None
```

Comment:
NIT: maybe with `elif len(trainer.loggers) > 0` the assert here can be avoided

Reply:
That's right, modified.

I didn't think much about it at the time and thought to keep it as close to the original code as possible :)